tools/validate_configs/validate_golden_copy.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+run_test() {
+	bp=$1
+	gc=$2
+	tmpdir="$(mktemp -d)"
+	bpFile=$(basename "$bp")
+	DEPLOYMENT="golden_copy_deployment"
+	PROJECT="invalid-project"
+	VALIDATORS_TO_SKIP="test_project_exists,test_apis_enabled,test_region_exists,test_zone_exists,test_zone_in_region"
+	GHPC_PATH="${cwd}/ghpc"
+	# Cover the three possible starting sequences for local sources: ./ ../ /
+	LOCAL_SOURCE_PATTERN='source:\s\+\(\./\|\.\./\|/\)'
+
+	echo "testing ${bp} in ${tmpdir} against ${gc}"
+	cp "${bp}" "${tmpdir}/"
+
+	# Only run from the repo directory if there are local modules, otherwise
+	# run the test from the test directory using the installed ghpc binary.
+	if grep -q "${LOCAL_SOURCE_PATTERN}" "${cwd}/${bp}"; then
+		cd "${cwd}"
+	else
+		cd "${tmpdir}"
+	fi
+	${GHPC_PATH} create -l ERROR \
+		--skip-validators="${VALIDATORS_TO_SKIP}" \
+		--vars="project_id=${PROJECT},deployment_name=${DEPLOYMENT}" \
+		"${tmpdir}"/"${bpFile}" >/dev/null ||
+		{
+			echo "*** ERROR: error creating deployment with ghpc for ${bpFile}"
+			exit 1
+		}
+	if grep -q "${LOCAL_SOURCE_PATTERN}" "${cwd}/${bp}"; then
+		mv "${DEPLOYMENT}" "${tmpdir}"
+	fi
+	cd "${tmpdir}"/"${DEPLOYMENT}" || {
+		echo "*** ERROR: can't cd into the deployment folder ${DEPLOYMENT}"
+		exit 1
+	}
+
+	# Sanitize deployment folder
```

Comment:
it might make sense to TRAP the cleanup. e.g. (first link I found). do you agree?

https://www.linuxjournal.com/content/use-bash-trap-statement-cleanup-temporary-files

Reply:
I would prefer to consider it next time. This script is copy-paste of validate_config, we can change them both at once later.